packages/get-platform/src/getPlatform.ts

```diff
@@ -132,20 +133,26 @@ export function parseOpenSSLVersion(input: string): string | undefined {
 }
 
 // getOpenSSLVersion returns the OpenSSL version excluding the patch version, e.g. "1.1.x"
-export async function getOpenSSLVersion(): Promise<string | undefined> {
+export async function getOpenSSLVersion(platform: NodeJS.Platform, arch: Arch): Promise<string | undefined> {
+  let prefix: string | undefined = undefined
```

Comment:
What is that prefix for exactly? Can we give it a better, more descriptive name?

Reply:
![Screenshot_20220419-132438_Termux](https://user-images.githubusercontent.com/80613396/163986689-fd3f4bac-802d-4751-9e7c-d8c53a4f2ce7.jpg)

Its for the termux's root

Reply:
Isn't that then termux specific, and not Android specific actually?

Reply:
Yes

Reply:
Shouldn't we call this new "platform" something like `android-termux` then at least? I think calling it `android` might misrepresent the situation.

Reply:
For example next-swc uses android sdk/ndk to provide binaries for android

```yml
          - host: ubuntu-latest
            target: aarch64-linux-android
            build: |
              export CARGO_TARGET_AARCH64_LINUX_ANDROID_LINKER="${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android24-clang"
              export CC="${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android24-clang"
              export CXX="${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android24-clang++"
              export PATH="${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64/bin:${PATH}"
              npm i -g @napi-rs/cli@2.4.4 turbo@1.0.28
              turbo run build-native --cache-dir=".turbo" -- --release --target aarch64-linux-android
              ${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android-strip packages/next-swc/native/next-swc.*.node
          - host: ubuntu-latest
            target: armv7-linux-androideabi
            build: |
              export CARGO_TARGET_ARMV7_LINUX_ANDROIDEABI_LINKER="${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64/bin/armv7a-linux-androideabi24-clang"
              export CC="${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64/bin/armv7a-linux-androideabi24-clang"
              export CXX="${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64/bin/armv7a-linux-androideabi24-clang++"
              export PATH="${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64/bin:${PATH}"
              npm i -g @napi-rs/cli@2.4.4 turbo@1.0.28
              turbo run build-native --cache-dir=".turbo" -- --release --target armv7-linux-androideabi
              ${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/linux-x86_64/bin/arm-linux-androideabi-strip packages/next-swc/native/next-swc.*.node
          - host: ubuntu-latest
            target: 'aarch64-unknown-linux-musl'
            docker: ghcr.io/napi-rs/napi-rs/nodejs-rust:lts-alpine
            build: >-
              set -e &&
              npm i -g @napi-rs/cli@2.4.4 turbo@1.0.28 &&
              rustup toolchain install nightly-2022-02-23 &&
              rustup default nightly-2022-02-23 &&
              rustup target add aarch64-unknown-linux-musl &&
              turbo run build-native --cache-dir=".turbo" -- --release --target aarch64-unknown-linux-musl &&
              llvm-strip -x packages/next-swc/native/next-swc.*.node

```

Reply:
Then it actually looks like this to me:

- `os.platform()` returns `android`, so that is the value of `platform` we can work with. That is fine.
- What you are trying to achieve is that on `platform == android` it downloads the correct image that could work there. I think I see that change above, line 55/56.
- The changes here in `getOpenSSLVersion` will actually only work when Termux is installed, so that is something to be made clearer. Looking at the original code, it seems that before it was essentially a list of paths we tried to list and then grep for `ssl`, and what needs to be done is to essentially replace that list with another, Android+Termux specific one. If there are other uses of Prisma on Android, it would probably require an additional path to be checked. (As currently it is not even downloading the correct engine file, these additoinal tasks are of course fine to leave as a TODO in the code only.)
- What is left to check then is if `resolveDistro` actually also returns the correct value for it to download the correct files that would work on Android(with Termux).

Does this make sense to you?